met-api/src/met_api/models/engagement.py

```diff
@@ -108,3 +109,21 @@ def update_engagement(cls, engagement: EngagementSchema) -> DefaultMethodResult:
         query.update(update_fields)
         db.session.commit()
         return DefaultMethodResult(True, 'Engagement Updated', engagement_id)
+
+    @classmethod
+    def close_engagements_due(cls) -> list[EngagementSchema]:
+        """Update engagement to closed."""
+        engagements_schema = EngagementSchema(many=True)
+        update_fields = dict(
+            status_id=Status.Published,
+            updated_date=datetime.utcnow(),
+            updated_by=SYSTEM_USER
+        )
+        query = Engagement.query\
+            .filter(Engagement.status_id == Status.Published and Engagement.end_date < datetime.utcnow())
+        records = query.all()
+        if not records:
+            return []
+        query.update(update_fields)
+        db.session.commit()
+        return engagements_schema.dump(records)
```

Comment:
@saravanpa-aot I believe you made a point against returning schemas for models?

Reply:
Yeah,..we don't need to..We will handle it as a techdebt..But good to change it for new code